Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/AssetPlatformComponentRemover.cpp

```diff
@@ -54,8 +54,8 @@ namespace AzToolsFramework::Prefab::PrefabConversionUtils
                     {
                         // Loop over an entity's components backwards and pop-off components that shouldn't exist.
                         AZStd::vector<AZ::Component*> components = entity->GetComponents();
-                        const int oldComponentCount = components.size();
-                        for (int i = oldComponentCount - 1; i >= 0; --i)
+                        size_t oldComponentCount = components.size();
+                        for (size_t i = oldComponentCount - 1; i > 0; --i)
```

Comment:
`i' needs to be initialized to `oldComponentCount`.

Because when `oldComponentCount` is 1, the components will not be looped over as the condition `i > 0` fails.

However changing the condition to `i >=0` is not correct because that would lead to an infinite loop,

So the solution is to change the variable init of `i` to be the total count size and then inside the loop code perform the subtraction of `i -1`